### PR TITLE
Feature lwdev 3901 asset service client provide classes with radis

### PR DESCRIPTION
**Pull request by NovichikhinAlexey**

### CODE REVIEW

File: src/Lykke.Service.Assets/Cache/AssetsForClientCacheManager.cs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        public async Task SaveSwiftDepositEnabledForClient(string clientId, bool isIosDevice, bool swiftDepositEnabled)
+        {
+            try
+            {
+                await _cache.SetStringAsync(GetKeySwiftDepositEnabled(clientId, isIosDevice), swiftDepositEnabled.ToJson(),
+                    new DistributedCacheEntryOptions() { AbsoluteExpirationRelativeToNow = _settings.AssetsForClientCacheTimeSpan });
+            }
+            catch (Exception ex)
+            {
+                await _log.WriteErrorAsync(nameof(AssetsForClientCacheManager), nameof(SaveCashInViaBankCardEnabledForClient), clientId, ex);
```
Comment (aushakov) [inline src/Lykke.Service.Assets/Cache/AssetsForClientCacheManager.cs]: Second parameter should be nameof(SaveSwiftDepositEnabledForClient)

File: src/Lykke.Service.Assets/Cache/AssetsForClientCacheManager.cs [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+            {
+                var json = await _cache.GetStringAsync(GetKeyAvailableAssets(clientId, isIosDevice));
+                if (!string.IsNullOrEmpty(json))
+                {
+                    return json.DeserializeJson<List<string>>();
+                }
+            }
+            catch (Exception ex)
+            {
+                await _log.WriteErrorAsync(nameof(AssetsForClientCacheManager), nameof(SaveCashInViaBankCardEnabledForClient), clientId, ex);
```
Comment (aushakov) [inline src/Lykke.Service.Assets/Cache/AssetsForClientCacheManager.cs]: Second parameter should be nameof(TryGetAssetForClient)

File: src/Lykke.Service.Assets/Cache/AssetsForClientCacheManager.cs [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+            {
+                var json = await _cache.GetStringAsync(GetKeyCashInViaBankCardEnabled(clientId, isIosDevice));
+                if (!string.IsNullOrEmpty(json))
+                {
+                    return json.DeserializeJson<bool>();
+                }
+            }
+            catch (Exception ex)
+            {
+                await _log.WriteErrorAsync(nameof(AssetsForClientCacheManager), nameof(SaveCashInViaBankCardEnabledForClient), clientId, ex);
```
Comment (aushakov) [inline src/Lykke.Service.Assets/Cache/AssetsForClientCacheManager.cs]: Second parameter should be nameof(TryGetSaveCashInViaBankCardEnabledForClient)

File: src/Lykke.Service.Assets/Cache/AssetsForClientCacheManager.cs [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+            {
+                var json = await _cache.GetStringAsync(GetKeySwiftDepositEnabled(clientId, isIosDevice));
+                if (!string.IsNullOrEmpty(json))
+                {
+                    return json.DeserializeJson<bool>();
+                }
+            }
+            catch (Exception ex)
+            {
+                await _log.WriteErrorAsync(nameof(AssetsForClientCacheManager), nameof(SaveCashInViaBankCardEnabledForClient), clientId, ex);
```
Comment (aushakov) [inline src/Lykke.Service.Assets/Cache/AssetsForClientCacheManager.cs]: Second parameter should be nameof(TryGetSaveSwiftDepositEnabledForClient)

File: src/Lykke.Service.Assets/Cache/AssetsForClientCacheManager.cs [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+            }
+            return null;
+        }
+
+        private string GetKeyAvailableAssets(string clientId, bool isIosDevice)
+        {
+            return string.Format(":Assets:Client:Available_Assets:{0}_{1}", clientId, isIosDevice ? "ios_device" : "");
```
Comment (aushakov) [inline src/Lykke.Service.Assets/Cache/AssetsForClientCacheManager.cs]: Could you use interpolated string here and below?

File: src/Lykke.Service.Assets/Startup.cs [hunk 6]
```diff
@@ -92,12 +94,40 @@ public IServiceProvider ConfigureServices(IServiceCollection services)
...
+                    options.Configuration = settings.CurrentValue.AssetsService.RadisSettings.RedisConfiguration;
+                    options.InstanceName = settings.CurrentValue.AssetsService.RadisSettings.InstanceName;
+                });
+
                 var builder = new ContainerBuilder();
 
+
+                var redis = ConnectionMultiplexer.Connect(settings.CurrentValue.AssetsService.RadisSettings.RedisConfiguration);
+
+                builder.RegisterInstance(redis).SingleInstance();
```
Comment (aushakov) [inline src/Lykke.Service.Assets/Startup.cs]: Is it possible to move registration to appropriated autofac module?

File: src/Lykke.Service.Assets.Services/AssetGroupService.cs [hunk 7]
```diff
@@ -213,6 +255,10 @@ public class AssetGroupService : IAssetGroupService
                 await _clientAssetGroupLinkRepository.AddAsync(assetClientGroupLink);
                 await _assetGroupClientLinkRepository.AddAsync(assetGroupClientLink);
             }
+
+            await _cacheManager.RemoveClientFromChache(clientId);
+
+            await _cacheManager.RemoveClientFromChache(clientId);
```
Comment (aushakov) [inline src/Lykke.Service.Assets.Services/AssetGroupService.cs]: Remove duplicate

File: src/Lykke.Service.Assets/Cache/AssetsForClientCacheManager.cs [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+{
+    public class AssetsForClientCacheManager : IAssetsForClientCacheManager
+    {
+        private readonly IDistributedCache _cache;
+        private readonly IAssetsForClientCacheManagerSettings _settings;
+        private readonly StackExchange.Redis.IServer _redisServer;
+        private readonly StackExchange.Redis.IDatabase _redisDatabase;
+        private readonly ILog _log;
+
+        public AssetsForClientCacheManager(IDistributedCache cache, IAssetsForClientCacheManagerSettings settings,
```
Comment (aushakov) [inline src/Lykke.Service.Assets/Cache/AssetsForClientCacheManager.cs]: It should be better parameter per line. Future mergers will be easy.

File: src/Lykke.Service.Assets/Cache/AssetsForClientCacheManager.cs (hunk 5 above)
Comment (NovichikhinAlexey) [reply src/Lykke.Service.Assets/Cache/AssetsForClientCacheManager.cs]: it's same, but this paramert not simple 
isIosDevice ? "ios_device" : ""

Comment (aushakov) [reply src/Lykke.Service.Assets/Cache/AssetsForClientCacheManager.cs]: Seems good to me
return $":Assets:Client:Available_Assets:{clientId}_{(isIosDevice ? "ios_device" : string.Empty)}";

**Pull request merged**